### PR TITLE
Pvs/v560

### PR DESCRIPTION
**Pull request by ngortheone**
Fixes 2 [V560](https://www.viva64.com/en/w/v560/) in screen.c

### CODE REVIEW

File: src/nvim/screen.c [hunk 1]
```diff
@@ -3805,8 +3804,7 @@ win_line (
           && ((area_attr != 0 && vcol == fromcol
                && (VIsual_mode != Ctrl_V
                    || lnum == VIsual.lnum
-                   || lnum == curwin->w_cursor.lnum)
-               && c == NUL)
```
Comment (ngortheone) [inline src/nvim/screen.c]: c == NUL is always true. See line 3780

File: src/nvim/screen.c [hunk 2]
```diff
@@ -4033,7 +4031,7 @@ win_line (
         && filler_todo <= 0
         && (wp->w_p_rl ? col == 0 : col == grid->Columns - 1)
         && (*ptr != NUL
-            || (wp->w_p_list && lcs_eol_one > 0)
```
Comment (ngortheone) [inline src/nvim/screen.c]: wp->w_p_list is already checked on line 4031

**Comment by ngortheone** [comment]
@justinmk 
I am tired resubmitting this PR for CI testing. Every time something new fails... This PR is not breaking any stable tests

**Comment by justinmk** [comment]
There's no need to resubmit anything if only flaky test is failing.

File: src/nvim/screen.c (hunk 1 above)
Comment (justinmk) [reply src/nvim/screen.c]: I would usually add `assert(c != NUL);` just to be safe. In this case it's pretty obvious that the entire block is predicated on that fact, but even then ...

**Pull request merged**